ignite/cmd/network.go

```diff
@@ -68,46 +65,49 @@ func NewNetwork() *cobra.Command {
 
 var cosmos *cosmosclient.Client
 
-type NetworkBuilder struct {
-	AccountRegistry cosmosaccount.Registry
-	Spinner         *clispinner.Spinner
+type (
+	NetworkBuilderOption func(builder *NetworkBuilder)
 
-	ev  events.Bus
-	wg  *sync.WaitGroup
-	cmd *cobra.Command
-	cc  cosmosclient.Client
-}
+	NetworkBuilder struct {
+		AccountRegistry cosmosaccount.Registry
 
-func newNetworkBuilder(cmd *cobra.Command) (NetworkBuilder, error) {
-	var err error
+		ev  events.Bus
+		cmd *cobra.Command
+		cc  cosmosclient.Client
+	}
+)
 
-	n := NetworkBuilder{
-		Spinner: clispinner.New(),
-		ev:      events.NewBus(),
-		wg:      &sync.WaitGroup{},
-		cmd:     cmd,
+func CollectEvents(ev events.Bus) NetworkBuilderOption {
+	return func(builder *NetworkBuilder) {
+		builder.ev = ev
 	}
+}
 
-	n.wg.Add(1)
-	go printEvents(n.wg, n.ev, n.Spinner)
+func newNetworkBuilder(cmd *cobra.Command, options ...NetworkBuilderOption) (NetworkBuilder, error) {
+	var (
+		err error
+		n   = NetworkBuilder{cmd: cmd}
+	)
 
 	if n.cc, err = getNetworkCosmosClient(cmd); err != nil {
-		n.Cleanup()
 		return NetworkBuilder{}, err
 	}
 
 	n.AccountRegistry = n.cc.AccountRegistry
 
+	for _, apply := range options {
+		apply(&n)
+	}
 	return n, nil
 }
 
 func (n NetworkBuilder) Chain(source networkchain.SourceOption, options ...networkchain.Option) (*networkchain.Chain, error) {
-	options = append(options, networkchain.CollectEvents(n.ev))
-
 	if home := getHome(n.cmd); home != "" {
 		options = append(options, networkchain.WithHome(home))
 	}
 
+	options = append(options, networkchain.CollectEvents(n.ev))
```

Comment:
Can we avoid passing this option of n.ev is nil? And same goes for the Chain()

Reply:
You mean to not provide n.ev to options if n.ev is nil?

Reply:
yes

Reply:
Actually n.ev is not a pointer so it can't be nil. If it was not initialised default value struct will correctly handle method calls on it